crabs/tracker/evaluate_tracker.py

```diff
@@ -131,153 +150,200 @@ def calculate_iou(self, box1: np.ndarray, box2: np.ndarray) -> float:
 
     def count_identity_switches(
         self,
-        prev_frame_ids: Optional[list[list[int]]],
-        current_frame_ids: Optional[list[list[int]]],
+        gt_to_tracked_id_previous_frame: Optional[Dict[int, int]],
+        gt_to_tracked_id_current_frame: Dict[int, int],
     ) -> int:
         """
         Count the number of identity switches between two sets of object IDs.
 
         Parameters
         ----------
-        prev_frame_ids : Optional[list[list[int]]]
-            List of object IDs in the previous frame.
-        current_frame_ids : Optional[list[list[int]]]
-            List of object IDs in the current frame.
+        gt_to_tracked_id_previous_frame : Optional[Dict[int, int]]
+            A dictionary mapping ground truth IDs to predicted IDs from the previous frame.
+        gt_to_tracked_id_current_frame : Dict[int, int]
+            A dictionary mapping ground truth IDs to predicted IDs for the current frame.
 
         Returns
         -------
         int
             The number of identity switches between the two sets of object IDs.
         """
-
-        if prev_frame_ids is None or current_frame_ids is None:
+        if gt_to_tracked_id_previous_frame is None:
             return 0
 
-        # Initialize count of identity switches
-        num_switches = 0
-
-        prev_ids = set(prev_frame_ids[0])
-        current_ids = set(current_frame_ids[0])
-
-        # Calculate the number of switches by finding the difference in IDs
-        num_switches = len(prev_ids.symmetric_difference(current_ids))
-
-        return num_switches
+        switch_counter = 0
+
+        # Compute sets of ground truth IDs for current and previous frames
+        gt_ids_current_frame = set(gt_to_tracked_id_current_frame.keys())
+        gt_ids_prev_frame = set(gt_to_tracked_id_previous_frame.keys())
+
+        # Compute lists of ground truth IDs that continue, disappear, and appear
+        gt_ids_cont = list(gt_ids_current_frame & gt_ids_prev_frame)
+        gt_ids_disappear = list(gt_ids_prev_frame - gt_ids_current_frame)
+        gt_ids_appear = list(gt_ids_current_frame - gt_ids_prev_frame)
+
+        # Store used predicted IDs to avoid double counting
```

Comment:
```suggestion
        # Store used predicted IDs to avoid double counting
        # In `used_pred_ids` we log IDs from either the current or the previous frame that have been involved in an already counted ID switch.
```